src/main/java/edu/umich/tl/ApiCallHandler.java

```diff
@@ -112,7 +187,7 @@ private HttpResponse apiDirectCanvas(String url) {
 		return httpResponse;
```

Comment:
apiDirectCanvas can return null if there is an exception in client.execute
and that would propagate back up.

On Wed, Oct 7, 2015 at 10:58 AM, Pushyami Gundala <notifications@github.com>
wrote:

> In src/main/java/edu/umich/tl/ApiCallHandler.java
> <https://github.com/tl-its-umich-edu/canvasCourseDeleteUtility/pull/5#discussion_r41400648>
> :
>
> > @@ -112,7 +187,7 @@ private HttpResponse apiDirectCanvas(String url) {
> >  		return httpResponse;
>
> httpResponse would never be null. It would have a response( good or bad)
> or some run time exception can occur. is the confusion is because of
> checkResponseState(). here the response is not httpResponse but a list
>
> private static boolean checkResponseState(List<HashMap<String, Object>> response) {
>         //response=null is a error condition so we don't want to delete the course just skip the course from the delete determination logic.
>         if(response==null || (!response.isEmpty())) {
>             return true;
>         }
>         return false;
>     }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tl-its-umich-edu/canvasCourseDeleteUtility/pull/5/files#r41400648>
> .
>